drbd_metrics.go

```diff
@@ -111,3 +120,28 @@ func parseDrbdStatus(statusRaw []byte) ([]drbdStatus, error) {
 	}
 	return drbdDevs, nil
 }
+
+func (c *drbdCollector) setDrbdSplitBrainMetric(ch chan<- prometheus.Metric) {
+
+	// set split brain metric
+	// by default if the custom hook is not set, the exporter will not be able to detect it
+	files, err := ioutil.ReadDir(c.drbdSplitBrainPath)
+	if err != nil {
+		log.Warnf("Error while reading directory /var/run/drbd/splitbrain: %s", err)
+	}
+
+	for _, f := range files {
+		// check if in directory there are file of syntax we expect (nil is when there is not any)
+		match, _ := filepath.Glob(c.drbdSplitBrainPath + "/drbd-split-brain-detected-*")
+		if match != nil {
+			resAndVolume := strings.Split(f.Name(), "drbd-split-brain-detected-")[1]
+			// avoid to have index out range panic error (in case the there is not resource-volume syntax)
+			if len(strings.Split(resAndVolume, "-")) == 2 {
+				resource := strings.Split(resAndVolume, "-")[0]
+				volume := strings.Split(resAndVolume, "-")[1]
+				ch <- c.makeGaugeMetric("split_brain", float64(1), resource, volume)
+				log.Warnf("Found split brain for drbd %s %s! Please remove file once you have resolved splitbrain", resource, volume)
+			}
+		}
```

Comment:
Here is what I meant by "fail fast, fail early":

Instead of:
```go
if match != nil {
   // happy path
}
```
consider:
```go
if match == nil {
    continue
}
// happy path
```

Same goes for the length check in the string split.
Talking about the string split, we are wastefully calling the same split over and over: I think we should rather store the result of the split in a variable:

```go
split := strings.Split(resAndVolume, "-")
if len(split) != 2 {
    // maybe log a warning
    continue
}
// etc
```

I also don't think we should log a warning about the system being monitored, we should only log the status of the exporter itself!

Reply:
make sense done